keras/applications/regnet.py

```diff
@@ -0,0 +1,1349 @@
+        `classifier_activation=None` to return the logits of the "top" layer.
+        Defaults to "softmax".
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+
+  Returns:
+    A `keras.Model` instance.
+"""
+
+def PreStem():
+  """Rescales and normalizes inputs to [0,1] and ImageNet mean and std.
+  
+  Args:
+    x: input tensor
+    
+  Returns:
+    Rescaled and normalized tensor
+  """
+
+  def apply(x):
+    IMGNET_MEAN = tf.constant([0.485, 0.456, 0.406])
+
+    IMGNET_STD = tf.constant([0.229, 0.224, 0.225])
+
+    x = layers.Rescaling(scale=1./255., name="prestem_rescaling")(x)
+    x = layers.Normalization(mean=IMGNET_MEAN, 
+          variance=tf.math.square(IMGNET_STD), name="prestem_normalization")(x)
+
+    return x
+  return apply
+
+def Stem():
+  """Implementation of RegNet stem. (Common to all model variants)
+  
+  Args:
+    x: Input tensor. Should be 224x224, rescaled and normalized to [0,1].   
+
+  Returns:
+    Output tensor of the Stem
+  """
+  def apply(x):
+    x = layers.Conv2D(32, (3, 3), strides=2, use_bias=False, name="stem_conv")(x)
+    x = layers.BatchNormalization(momentum=0.9, epsilon=1e-5, name="stem_bn")(x)
+    x = layers.ReLU(name="stem_relu")(x)
+
+    return x
+  return apply
+
+
+def XBlock(filters_in,
+           filters_out,
+           group_width,
+           stride=1,
+           name=""):
+  """Implementation of X Block. 
+  Reference: [Designing Network Design Spaces](https://arxiv.org/abs/2003.13678)
+
+  Args:
+    inputs: input tensor
+    filters_in: filters in the input tensor
+    filters_out: filters in the output tensor
+    group_width: group width
+    stride: stride
+    name: name prefix
+
+  Return:
+    Output tensor of the block 
+  """
+  def apply(inputs):
+    if filters_in != filters_out and stride == 1:
+      raise ValueError("""Input filters and output filters are not equal for stride
+                          1. Please check inputs and try again.""")
+
+    # Declare layers
+    groups = filters_out // group_width
+    
+    if stride != 1:
+      skip = layers.Conv2D(filters_out, (1, 1),
+                          strides=stride, use_bias=False,
+                          name=name + "_skip_1x1")(inputs)
+      skip = layers.BatchNormalization(
+          momentum=0.9, epsilon=1e-5, name=name + "_skip_bn")(skip)
+      conv_3x3 = layers.Conv2D(filters_out, (3, 3), use_bias=False, strides=stride,
+                              groups=groups, name=name + "_conv_3x3")
+    else:
+      skip = inputs
+      conv_3x3 = layers.Conv2D(filters_out, (3, 3), use_bias=False,
+                              groups=groups, name=name + "_conv_3x3")
+    
+      # Build block
+    # conv_1x1_1
+    x = layers.Conv2D(filters_out, (1, 1), use_bias=False,
+                      name=name + "_conv_1x1_1")(inputs)
+    x = layers.BatchNormalization(
+        momentum=0.9, epsilon=1e-5, name=name + "_conv_1x1_1_bn")(x)
```

Comment:
If these are not default values of the `BatchNormalization` layer, then it would make sense to note where these values are coming from. Applies to other similar cases as well.

Reply:
Noted at the start of the file.